targets/linux/main.c

```diff
@@ -73,173 +165,168 @@ bool run_test(const char *filename) {
+
+  filelist_foreach(fl, file) { run_memory_test(file, vars); }
+
   return true;
 }
 
-
-void sig_handler(int sig)
-{
-  //printf("Got Signal %d\n",sig);fflush(stdout);
-  if (sig==SIGINT)
+void sig_handler(int sig) {
+  // warning("Got Signal %d\n",sig);fflush(stdout);
+  if (sig == SIGINT)
     jspSetInterrupted(true);
 }
 
 void show_help() {
-    printf("Usage:\n");
-    printf("   ./espruino                           : JavaScript immediate mode (REPL)\n");
-    printf("   ./espruino script.js                 : Load and run script.js\n");
-    printf("   ./espruino -e \"print('Hello World')\" : Print 'Hello World'\n");
-    printf("\n");
-    printf("Options:\n");
-    printf("   -h, --help              Print this help screen\n");
-    printf("   -e, --eval script       Evaluate the JavaScript supplied on the command-line\n");
+  warning("Usage:");
+  warning("   ./espruino                           : JavaScript immediate mode "
+          "(REPL)");
+  warning("   ./espruino script.js                 : Load and run script.js");
+  warning("   ./espruino -e \"print('Hello World')\" : Print 'Hello World'");
+  warning("");
+  warning("Options:");
+  warning("   -h, --help              Print this help screen");
+  warning("   -e, --eval script       Evaluate the JavaScript supplied on the "
+          "command-line");
 #ifdef USE_TELNET
-    printf("   --telnet                Enable internal telnet server on port 2323\n");
+  warning(
+      "   --telnet                Enable internal telnet server on port 2323");
 #endif
-    printf("   --test-all              Run all tests (in 'tests' directory)\n");
-    printf("   --test test.js          Run the supplied test\n");
-    printf("   --test-mem-all          Run all Exhaustive Memory crash tests\n");
-    printf("   --test-mem test.js      Run the supplied Exhaustive Memory crash test\n");
-    printf("   --test-mem-n test.js #  Run the supplied Exhaustive Memory crash test with # vars\n");
+  warning("   --test-all              Run all tests (in 'tests' directory)");
+  warning("   --test-dir dir          Run all tests in directory 'dir'");
+  warning("   --test test.js          Run the supplied test");
+  warning("   --test-mem-all          Run all Exhaustive Memory crash tests");
+  warning("   --test-mem test.js      Run the supplied Exhaustive Memory crash "
+          "test");
+  warning("   --test-mem-n test.js #  Run the supplied Exhaustive Memory crash "
+          "test with # vars");
 }
 
 void die(const char *txt) {
-  printf("%s", txt);
+  warning("%s", txt);
   exit(1);
 }
 
 int handleErrors() {
   int e = 0;
   JsVar *exception = jspGetException();
   if (exception) {
-    jsiConsolePrintf("Uncaught %v\n", exception);
+    jsiConsolePrintf("Uncaught %v", exception);
     jsvUnLock(exception);
     e = 1;
   }
 
   if (jspIsInterrupted()) {
     jsiConsoleRemoveInputLine();
-    jsiConsolePrint("Execution Interrupted.\n");
+    jsiConsolePrint("Execution Interrupted.");
```

Comment:
Is there a reason for this? jsiConsolePrint doesn't add a newline so I'd be concerned this would cause issues